### PR TITLE
Use filepath normalizing for blacklist

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Comment by Divran** [comment]
Do any of these do the same thing as the function you implemented: 
http://wiki.garrysmod.com/page/util/RelativePathToFull
http://wiki.garrysmod.com/page/string/GetPathFromFilename

**Comment by thegrb93** [comment]
Doesn't seem like those functions do this.

File: lua/entities/gmod_wire_expression2/core/e2lib.lua [hunk 1]
```diff
@@ -47,17 +47,12 @@ function E2Lib.setAng(ent, ang)
...
-	"pp[%./\\]+copy"
+	["pp/copy"] = true
 }
 local function validMaterial(material)
-	local lower = string.lower(material)
-	for _, v in ipairs(material_blacklist) do
-		if string.find(lower, v) then return "" end
-	end
+	local path = string.StripExtension(string.GetNormalizedFilepath(string.lower(material)))
+	if material_blacklist[path] then return "" end
```
Comment (CaptainPRICE) [inline lua/entities/gmod_wire_expression2/core/e2lib.lua]: ew my eyes. Could you change this to `return material_blacklist[path] and "" or material`
Thanks.

Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/e2lib.lua]: Ternary ops are slower in lua bb.

Comment (CaptainPRICE) [inline lua/entities/gmod_wire_expression2/core/e2lib.lua]: GMod has LuaJIT .^.

**Comment by thegrb93** [comment]
I'll fix it

**Comment by thegrb93** [comment]
I think that fixes it

**Pull request merged**